core/src/main/scala/org/apache/spark/deploy/security/HBaseDelegationTokenProvider.scala

```diff
@@ -54,7 +54,8 @@ private[security] class HBaseDelegationTokenProvider
       creds.addToken(token.getService, token)
     } catch {
       case NonFatal(e) =>
-        logWarning(Utils.createFailedToGetTokenMessage(serviceName, e) + log" Retrying to fetch " +
+        logWarning(log"${MDC(MESSAGE, Utils.createFailedToGetTokenMessage(serviceName, e))} " +
```

Comment:
createFailedToGetTokenMessage already returns `MessageWithContext`. Otherwise, it can't concat with other `MessageWithContext`